File: bubble_sort_10000.rb
def bubble_sort(array)
  n = array.length
  loop do
    swapped = false
    (n-1).times do |i|
      if array[i] > array[i+1]
        array[i], array[i+1] = array[i+1], array[i]
        swapped = true
      end
    end
    break unless swapped
  end
  array
end

# ランダムな10000個の数字の配列を生成
array = (1..10000).to_a.shuffle

# バブルソートで昇順に並び替え
sorted_array = bubble_sort(array)

# 結果を表示
puts sorted_array

# 実行結果
# ruby bubble_sort_10000.rb  4.77s user 0.04s system 99% cpu 4.825 total

File: bubble_sort_10000.c
#include <stdio.h>
#include <stdlib.h>

void bubbleSort(int array[], int size) {
  int i, j;
  for (i = 0; i < size - 1; i++) {
    for (j = 0; j < size - i - 1; j++) {
      if (array[j] > array[j + 1]) {
        int temp = array[j];
        array[j] = array[j + 1];
        array[j + 1] = temp;
      }
    }
  }
}

int main() {
  int size = 10000;
  int array[size];

  // ランダムな10000個の数字の配列を生成
  for (int i = 0; i < size; i++) {
    array[i] = i + 1;
  }

  // 配列をシャッフル
  for (int i = 0; i < size - 1; i++) {
    int j = i + rand() / (RAND_MAX / (size - i) + 1);
    int temp = array[j];
    array[j] = array[i];
    array[i] = temp;
  }

  // バブルソートで昇順に並び替え
  bubbleSort(array, size);

  // 結果を表示
  for (int i = 0; i < size; i++) {
    printf("%d ", array[i]);
  }
  printf("\n");

  return 0;
}

//実行結果
//./a.out  0.17s user 0.00s system 41% cpu 0.425 total